### PR TITLE
Javascript refactor

### PR DESCRIPTION
**Pull request by giuliojiang**
* split the javascript files

* add a timeout in Mysql connection java file, and reformat file to use spaces for indentation instead of a mix of tabs and spaces

### CODE REVIEW

**Review by AbraaoMota — APPROVED** [review]
Much better

**Review by procaconsul — APPROVED** [review]
Clean work.

File: frontend/webclient/script/professorcontroller.js [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+      }
+      $scope.editing_source_cache = acc;
+  }
+  
+  $scope.add_feedback_annotation = function(filename, lineno, text, oldannotation, data_entry) {
+      if (!$scope.editing_file) {
+          $scope.editing_data_entry = data_entry;
+          $scope.editing_file = filename;
+          $scope.editing_data_entry.editing_annotation = oldannotation;
+          $scope.editing_data_entry.editing_ann_type = "semantic";
```
Comment (helicopter88) [inline frontend/webclient/script/professorcontroller.js:200]: why semantic?

**Review by helicopter88 — APPROVED** [review]
The SQL changes should be in a separate set as they're not really related, but since it's very small I'll not give you a -1

File: frontend/webclient/script/professorcontroller.js (hunk 1 above)
Comment (AbraaoMota) [reply frontend/webclient/script/professorcontroller.js:200]: I think it's being used as a default value - probably better to have something like "Please write error type here"

Comment (giuliojiang) [reply frontend/webclient/script/professorcontroller.js:200]: it's a placeholder. It will be replaced by a drop-down menu

**Pull request merged**